File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {
    let s = new Set();
    let n = grid.length;
    let a = -1;
    let b = -1;
    let actSum = 0;
    let expSum = 0;
    for(let i=0; i<n; i++){
        for(let j=0; j<n; j++){
            actSum += grid[i][j];
            if(s.has(grid[i][j])){
                a = grid[i][j];
            }else{
                s.add(grid[i][j])
            }
        }
    }
    expSum = (n*n) * (n*n + 1) / 2;
    b = expSum - actSum + a;
    return [a, b]
};

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        s = set()
        a = 0
        actSum = 0
        expSum = 0
        b = 0
        n = len(grid)
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                actSum += grid[i][j]
                if (grid[i][j] in s):
                    a = grid[i][j]
                else:
                    s.add(grid[i][j])
        
        expSum = n*n * (n*n + 1) / 2

        b = int(expSum - actSum + a)
        return [a, b]